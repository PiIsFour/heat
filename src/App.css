.App {
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	/* grid-auto-rows: 1fr; */
	grid-auto-flow: dense;
	grid-gap: 2em;
	padding: 2em;
}

.App > .notice {
	background-color: #64CEAA;
	font-size: 1.5em;
	width: 50%;
	text-align: center;
	margin-top: 1em;
	padding-top: .3em;
	padding-bottom: .3em;
	border-radius: .3em;
	position: absolute;
	margin-left: auto;
	margin-right: auto;
	left: 0;
	right: 0;
}
