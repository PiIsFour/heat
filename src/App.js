import React, { Component } from 'react';
import './App.css';

import MeasureGroup from './MeasureGroup';
import composeState from './state';
import makeCalculator from './calculator';

class App extends Component {
  constructor(props){
    super(props);
    const state = composeState()
    .compose("P", "kW", 55, <mrow><mi>P</mi></mrow>)
    .compose("twin", "°C", 20, <mrow><msub><mi>T</mi><mi>W in</mi></msub></mrow>)
    .compose("twout", "°C", 30, <mrow><msub><mi>T</mi><mi>W out</mi></msub></mrow>)
    .compose("cw", "kJ/kgK", 4.18, <mrow><msub><mi>c</mi><mi>W</mi></msub></mrow>)
    .compose("Mw", "kg/s", 1, <mrow><msub><mi>M</mi><mi>W</mi></msub></mrow>)
    .compose("toin", "°C", 60, <mrow><msub><mi>T</mi><mi>O in</mi></msub></mrow>)
    .compose("toout", "°C", 40, <mrow><msub><mi>T</mi><mi>O out</mi></msub></mrow>)
    .compose("co", "kJ/kgK", 2, <mrow><msub><mi>c</mi><mi>O</mi></msub></mrow>)
    .compose("Mo", "kg/s", 1, <mrow><msub><mi>M</mi><mi>O</mi></msub></mrow>)
    .compose("A", "m²", 1, <mrow><mi>A</mi></mrow>)
    .compose("k", "kW/m²K", 1, <mrow><mi>k</mi></mrow>)
    .compose("tlog", "K", 1, <mrow><msub><mi>T</mi><mi>log</mi></msub></mrow>)
    .compose("alpha", "", 1, <mrow><mi>alpha</mi></mrow>)
    state.all = {
      valueChanged: (name, value) => {
        const state = {...this.state};
        state[name].value = value;
        this.calc.run(state);
        this.setState(state);
      },
      inputToggled: (name, input) => {
        const state = {...this.state};
        state[name].input = input;
        this.calc.run(state);
        this.setState(state);
      },
    };
    this.state = state;
		this.calc = makeCalculator()
    .add("P", ["twout", "twin", "cw", "Mw"], i => (i.twout-i.twin)*i.cw*i.Mw, <mrow><mi>P</mi><mo>=</mo><mo>(</mo><msub><mi>T</mi><mi>W out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W in</mi></msub><mo>)</mo><mo>*</mo><msub><mi>c</mi><mi>W</mi></msub><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow>)
    .add("P", ["toout", "toin", "co", "Mo"], i => (i.toin-i.toout)*i.co*i.Mo, <mrow><mi>P</mi><mo>=</mo><mo>(</mo><msub><mi>T</mi><mi>O out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>O in</mi></msub><mo>)</mo><mo>*</mo><msub><mi>c</mi><mi>O</mi></msub><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow>)
    .add("Mw", ["twout", "twin", "cw", "P"], i => i.P/(i.twout-i.twin)/i.cw, <mrow><msub><mi>M</mi><mi>W</mi></msub><mo>=</mo><mfrac><mi>P</mi><mrow><mo>(</mo><msub><mi>T</mi><mi>W out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W in</mi></msub><mo>)</mo><mo>*</mo><msub><mi>c</mi><mi>W</mi></msub></mrow></mfrac></mrow>)
    .add("Mo", ["toout", "toin", "co", "P"], i => i.P/(i.toin-i.toout)/i.co, <mrow><msub><mi>M</mi><mi>O</mi></msub><mo>=</mo><mfrac><mi>P</mi><mrow><mo>(</mo><msub><mi>T</mi><mi>O out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>O in</mi></msub><mo>)</mo><mo>*</mo><msub><mi>c</mi><mi>O</mi></msub></mrow></mfrac></mrow>)
    .add("cw", ["twout", "twin", "Mw", "P"], i => i.P/(i.twout-i.twin)/i.Mw, <mrow><msub><mi>c</mi><mi>W</mi></msub><mo>=</mo><mfrac><mi>P</mi><mrow><mo>(</mo><msub><mi>T</mi><mi>W out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W in</mi></msub><mo>)</mo><mo>*</mo><msub><mi>M</mi><mi>W</mi></msub></mrow></mfrac></mrow>)
    .add("co", ["toout", "toin", "Mo", "P"], i => i.P/(i.toin-i.toout)/i.Mo, <mrow><msub><mi>c</mi><mi>O</mi></msub><mo>=</mo><mfrac><mi>P</mi><mrow><mo>(</mo><msub><mi>T</mi><mi>O out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>O in</mi></msub><mo>)</mo><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow></mfrac></mrow>)
    .add("twout", ["twin", "cw", "Mw", "P"], i => i.twin+(i.P/i.cw/i.Mw), <mrow><msub><mi>T</mi><mi>W out</mi></msub><mo>=</mo><msub><mi>T</mi><mi>W in</mi></msub><mo>+</mo><mfrac><mi>P</mi><mrow><msub><mi>c</mi><mi>W</mi></msub><mo>*</mo><msub><mi>M</mi><mi>W</mi></msub></mrow></mfrac></mrow>)
    .add("twin", ["twout", "cw", "Mw", "P"], i => i.twout-(i.P/i.cw/i.Mw), <mrow><msub><mi>T</mi><mi>W in</mi></msub><mo>=</mo><msub><mi>T</mi><mi>W out</mi></msub><mo>-</mo><mfrac><mi>P</mi><mrow><msub><mi>c</mi><mi>W</mi></msub><mo>*</mo><msub><mi>M</mi><mi>W</mi></msub></mrow></mfrac></mrow>)
    .add("toin", ["toout", "co", "Mo", "P"], i => i.toout+(i.P/i.co/i.Mo), <mrow><msub><mi>T</mi><mi>O in</mi></msub><mo>=</mo><msub><mi>T</mi><mi>O out</mi></msub><mo>+</mo><mfrac><mi>P</mi><mrow><msub><mi>c</mi><mi>O</mi></msub><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow></mfrac></mrow>)
    .add("toout", ["toin", "co", "Mo", "P"], i => i.toin-(i.P/i.co/i.Mo), <mrow><msub><mi>T</mi><mi>O out</mi></msub><mo>=</mo><msub><mi>T</mi><mi>O in</mi></msub><mo>-</mo><mfrac><mi>P</mi><mrow><msub><mi>c</mi><mi>O</mi></msub><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow></mfrac></mrow>)
    .add("tlog", ["twin", "twout", "toin", "toout"], i => {
      if(Math.abs(i.toin-i.twout-i.toout+i.twin) < 0.1){
        return i.toin-i.twout;
      }
      return (i.toin-i.twout-i.toout+i.twin)/Math.log((i.toin-i.twout)/(i.toout-i.twin))
    }, <mrow><msub><mi>T</mi><mi>log</mi></msub><mo>=</mo><mfrac><mrow><msub><mi>T</mi><mi>O in</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>O out</mi></msub><mo>+</mo><msub><mi>T</mi><mi>W in</mi></msub></mrow><mrow><mo>log</mo><mo>(</mo><mfrac><mrow><msub><mi>T</mi><mi>O in</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W out</mi></msub></mrow><mrow><msub><mi>T</mi><mi>O out</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W in</mi></msub></mrow></mfrac><mo>)</mo></mrow></mfrac></mrow>)
    .add("P", ["A", "k", "tlog"], i => i.A*i.k*i.tlog, <mrow><mi>P</mi><mo>=</mo><mi>A</mi><mo>*</mo><mi>k</mi><mo>*</mo><msub><mi>T</mi><mi>log</mi></msub></mrow>)
    .add("A", ["P", "k", "tlog"], i => i.P/i.k/i.tlog, <mrow><mi>A</mi><mo>=</mo><mfrac><mi>P</mi><mrow><mi>k</mi><mo>*</mo><msub><mi>T</mi><mi>log</mi></msub></mrow></mfrac></mrow>)
    .add("k", ["P", "A", "tlog"], i => i.P/i.A/i.tlog, <mrow><mi>k</mi><mo>=</mo><mfrac><mi>P</mi><mrow><mi>A</mi><mo>*</mo><msub><mi>T</mi><mi>log</mi></msub></mrow></mfrac></mrow>)
    .add("alpha", ["A", "k", "co", "cw", "Mo", "Mw"], i => Math.exp(i.A*i.k*((1/i.co/i.Mo)-(1/i.cw/i.Mw))), <mrow><mi>alpha</mi><mo>=</mo><mo>exp</mo><mo>(</mo><mi>A</mi><mo>*</mo><mi>k</mi><mo>*</mo><mo>(</mo><mfrac><mn>1</mn><mrow><msub><mi>c</mi><mi>O</mi></msub><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow></mfrac><mo>-</mo><mfrac><mn>1</mn><mrow><msub><mi>c</mi><mi>W</mi></msub><mo>*</mo><msub><mi>M</mi><mi>W</mi></msub></mrow></mfrac><mo>)</mo><mo>)</mo></mrow>)
    .add("P", ["alpha", "toin", "twin", "co", "cw", "Mo", "Mw"], i => (i.alpha-1)*(i.toin-i.twin)/((i.alpha/i.co/i.Mo)-(1/i.cw/i.Mw)), <mrow><mi>P</mi><mo>=</mo><mfrac><mrow><mo>(</mo><mi>alpha</mi><mo>-</mo><mn>1</mn><mo>)</mo><mo>(</mo><msub><mi>T</mi><mi>O in</mi></msub><mo>-</mo><msub><mi>T</mi><mi>W in</mi></msub><mo>)</mo></mrow><mrow><mo>(</mo><mfrac><mi>alpha</mi><mrow><msub><mi>c</mi><mi>O</mi></msub><mo>*</mo><msub><mi>M</mi><mi>O</mi></msub></mrow></mfrac><mo>-</mo><mfrac><mn>1</mn><mrow><msub><mi>c</mi><mi>W</mi></msub><mo>*</mo><msub><mi>M</mi><mi>W</mi></msub></mrow></mfrac><mo>)</mo></mrow></mfrac></mrow>)
  }
  render() {
    const opt = [
      {name: "Wasser", childs: ["twin", "twout", "cw", "Mw"], style: {gridColumn: "3 / span 1", gridRowEnd: "span 2"}},
      {name: "Öl", childs: ["toin", "toout", "co", "Mo"], style: {gridColumn: "1 / span 1", gridRowEnd: "span 2"}},
      {name: "Leistung", childs: ["P"], style: {gridColumn: "2 / span 1"}},
      {name: "Kühler", childs: ["A", "k", "tlog", "alpha"], style: {gridColumn: "2 / span 1", gridRowEnd: "span 2"}},
    ];
    return (
      <div className="App">
        {opt.map(o => <MeasureGroup key={o.name} state={this.state} opt={o}/>)}
      </div>
    );
  }
}

export default App;
