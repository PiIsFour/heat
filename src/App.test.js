import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe('App', () => {  
  const component = shallow(
    <App />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
