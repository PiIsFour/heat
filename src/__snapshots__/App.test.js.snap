// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders and matches our snapshot 1`] = `
<div
  className="App"
>
  
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "2 / span 1",
      }
    }
  >
    <div
      className="name"
    >
      Leistung
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Leistung
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <mi>
              P
            </mi>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={55}
        />
        <span
          className="unit"
        >
          kW
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "1 / span 1",
        "gridRowEnd": "span 2",
      }
    }
  >
    <div
      className="name"
    >
      Öl
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Eingangstemperatur
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                T
              </mi>
              <mi>
                O in
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={60}
        />
        <span
          className="unit"
        >
          °C
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Ausgangstemperatur
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                T
              </mi>
              <mi>
                O out
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={40}
        />
        <span
          className="unit"
        >
          °C
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Massestrom
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                M
              </mi>
              <mi>
                O
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          kg/s
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Volumenstrom
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                Q
              </mi>
              <mi>
                O
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={60}
        />
        <span
          className="unit"
        >
          l/min
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "3 / span 1",
        "gridRowEnd": "span 2",
      }
    }
  >
    <div
      className="name"
    >
      Wasser
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Eingangstemperatur
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                T
              </mi>
              <mi>
                W in
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={20}
        />
        <span
          className="unit"
        >
          °C
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Ausgangstemperatur
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                T
              </mi>
              <mi>
                W out
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={30}
        />
        <span
          className="unit"
        >
          °C
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Massestrom
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                M
              </mi>
              <mi>
                W
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          kg/s
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Volumenstrom
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                Q
              </mi>
              <mi>
                W
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={60}
        />
        <span
          className="unit"
        >
          l/min
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "2 / span 1",
      }
    }
  >
    <div
      className="name"
    >
      Kühler
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Fläche
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <mi>
              A
            </mi>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          m²
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Wärmeübergangskoeffizient
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <mi>
              k
            </mi>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          kW/m²K
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "1 / span 1",
      }
    }
  >
    <div
      className="name"
    >
      Öl Konstanten
    </div>
    <div
      className="Measure"
      style={Object {}}
    >
      <div>
        Wärmekapazität
      </div>
      <div>
        <input
          checked={true}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                c
              </mi>
              <mi>
                O
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={false}
          onChange={[Function]}
          readOnly={false}
          type="text"
          value={2}
        />
        <span
          className="unit"
        >
          kJ/kgK
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={Object {}}
    >
      <div>
        Dichte
      </div>
      <div>
        <input
          checked={true}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                ρ
              </mi>
              <mi>
                O
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={false}
          onChange={[Function]}
          readOnly={false}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          kg/l
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "3 / span 1",
      }
    }
  >
    <div
      className="name"
    >
      Wasser Konstanten
    </div>
    <div
      className="Measure"
      style={Object {}}
    >
      <div>
        Wärmekapazität
      </div>
      <div>
        <input
          checked={true}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                c
              </mi>
              <mi>
                W
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={false}
          onChange={[Function]}
          readOnly={false}
          type="text"
          value={4.18}
        />
        <span
          className="unit"
        >
          kJ/kgK
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={Object {}}
    >
      <div>
        Dichte
      </div>
      <div>
        <input
          checked={true}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                ρ
              </mi>
              <mi>
                W
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={false}
          onChange={[Function]}
          readOnly={false}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          kg/l
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <div
    className="MeasureGroup"
    style={
      Object {
        "gridColumn": "2 / span 1",
      }
    }
  >
    <div
      className="name"
    >
      Automatische Größen
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Durchschnittstemperatur
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <msub>
              <mi>
                T
              </mi>
              <mi>
                log
              </mi>
            </msub>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          K
        </span>
      </div>
      <div>
        
      </div>
    </div>
    <div
      className="Measure"
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
    >
      <div>
        Alpha
      </div>
      <div>
        <input
          checked={false}
          className="input"
          disabled={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span>
          <math
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <mi>
              α
            </mi>
          </math>
        </span>
        <input
          className="value"
          disabled={true}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value={1}
        />
        <span
          className="unit"
        >
          
        </span>
      </div>
      <div>
        
      </div>
    </div>
  </div>
  <a
    aria-label="Open GitHub project"
    className="github-corner"
    href="https://github.com/PiIsFour/heat"
  >
    <svg
      height={80}
      style={
        Object {
          "fill": "#fff",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      viewBox="0 0 250 250"
      width={80}
    >
      <path
        className="octo-banner"
        d="M0 0l115 115h15l12 27 108 108V0z"
        fill="#64CEAA"
      />
      <path
        className="octo-arm"
        d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
        style={
          Object {
            "WebkitTransformOrigin": "130px 106px",
            "transformOrigin": "130px 106px",
          }
        }
      />
      <path
        className="octo-body"
        d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
      />
    </svg>
  </a>
</div>
`;
